Bot.Core/Announcer.cs

```diff
@@ -34,74 +34,42 @@ public Announcer(IServiceProvider sp)
             sp.Inject(out m_townDatabase);
             sp.Inject(out m_botSystem);
             sp.Inject(out m_botClient);
-            sp.Inject(out m_dateTime);            
+            sp.Inject(out m_dateTime);     
+            sp.Inject(out m_startupTownTasks);
 
-            var callbackFactory = sp.GetService<ICallbackSchedulerFactory>();
-            m_callbackScheduler = callbackFactory.CreateScheduler<Queue<TownKey>>(AnnounceToTowns, TimeSpan.FromMinutes(1));
-
-            m_botClient.Connected += OnClientConnected;
+            m_startupTownTasks.AddMaintenanceTask(AnnounceToTown);
         }
 
-        private async void OnClientConnected(object? sender, EventArgs e)
-        {
-            await AnnounceLatestVersion();
-        }
-
-        private async Task AnnounceToTowns(Queue<TownKey> towns)
+        private async Task AnnounceToTown(TownKey townKey)
         {
             bool restricted = bool.Parse(Environment.GetEnvironmentVariable("RESTRICT_ANNOUNCE") ?? "false");
```

Comment:
This would be better as:
```
if (!bool.TryParse(Environment.GetEnvironmentVariable("RESTRICT_ANNOUNCE"), out bool restricted))
    restricted = false;
```

Reply:
This should be using a member of type IEnvironment, not Environment directly, so it can be mocked

Reply:
This is purely a "when bot is deployed" protection feature though, so mocking it is... unneeded

Reply:
🤔 I'm trying to remember why I thought it was important to check the genuine environment var, not an arbitrary service that might or might not be the real env, for this